mempool/src/pool/mod.rs

```diff
@@ -1104,5 +1154,58 @@ impl<M: MemoryUsageEstimator> Mempool<M> {
     }
 }
 
+fn get_closest_value<V: Copy>(size_to_score: &BTreeMap<usize, V>, key: usize) -> Option<V> {
+    match (
+        size_to_score.range(..=key).next_back(),
+        size_to_score.range(key..).next(),
+    ) {
+        (Some((&k1, &v1)), Some((&k2, &v2))) => {
+            let diff1 = key - k1;
+            let diff2 = k2 - key;
+
+            if diff1 <= diff2 {
+                Some(v1)
+            } else {
+                Some(v2)
+            }
+        }
+        (Some((_, &v1)), None) => Some(v1),
+        (None, Some((_, &v2))) => Some(v2),
+        (None, None) => None,
+    }
+}
+
+fn generate_points(first: usize, last: usize, n: usize) -> Vec<usize> {
+    if first == last {
+        vec![first];
+    }
+
+    let mut points = Vec::with_capacity(n);
+    points.push(first);
+
+    let step = (last - first) / (n - 1);
+
+    for i in 1..(n - 1) {
+        points.push(first + step * i);
+    }
```

Comment:
This method will maximize off-by-one errors when `mod(last-first, n-1) == n-2`, but honestly I don't really care. Whatever works is fine since this is for interpolation.

However, please write tests for these simple functions.